docs/changelog/123188.yaml

```
@@ -0,0 +1,14 @@
+pr: 123188
+summary: Add deprecation warning to `TransportHandshaker`
+area: Infra/Core
+type: deprecation
+issues: []
+deprecation:
+  title: Add deprecation warning to `TransportHandshaker`
+  area: REST API
+  details: >
+    Users will now see a deprecation log when a remote node, who's version is < v8.18, performs a transport handshake
+    request with nodes >= v8.18 (e.g. during a CCS/CCR request). The remote cluster will need to be upgraded to >= v8.18
+    to maintain compatibility with clusters that have been upgraded to v9.
+  impact: >
+   Users should upgrade any clusters older than v8.18 to >= 8.18 to maintain compatibility with clusters upgraded to v9.
```

I think this should be imperative, and suggest "at least" over `>=`:

```suggestion
   Upgrade all of your clusters to at least 8.18.0 before upgrading any of them to 9.0.0 or later.
```